src/pages/datasets/index.tsx

```diff
@@ -1,74 +1,124 @@
+import {Plus} from '@/components/icons/plus'
+import {SearchBar} from '@/components/forms/searchbar'
+import Link from 'next/link'
 
 const Datasets: FunctionComponent = () => {
+  const {isLoading, data: datasetsData, error} = useQuery<IDataset[], Error>('datasets', fetchDatasets)
+  const {data: requests} = useQuery<IRequest[], Error>('requests', fetchRequests)
+  const [searchText, setSearchText] = useState('')
   const sections = [
-    {title: 'Permissions changes', value: 19, text: 'requests'},
-    {title: 'Budget changes', value: 5, text: 'requests'},
-    {title: 'Tensors pending deletion', value: 3, text: 'tensors'}
+    {
+      title: 'Permissions changes',
+      value: requests ? requests.filter(x => x.request_type === 'permissions' && x.status === 'pending').length : 0,
+      text: 'requests',
+      link: '/datasets/requests'
+    },
+    {title: 'Tensors pending deletion', value: 3, text: 'tensors', link: '/datasets/tensors'}
   ]
 
   const datasets = [
     {
-      title: 'Diabetes Study 01.289.301',
+      name: 'Diabetes Study 01.289.301',
       description:
         'This was a double-blind diabetes study done in coordination with UC Santa Barbara between July 1st, 2017 and January 1st, 2019.',
       tags: ['diabetes', 'california', 'healthcare', 'UCSF', 'beekeeper'],
       tensors: 2
     },
     {
-      title: 'Dementia MRI Scans (10k)',
+      name: 'Dementia MRI Scans (10k)',
       description:
         'Performed a database dump of exactly 10,000 patient records from our EMR. All patients were diagnosed with dementia within 10 years of the MRIs.',
       tags: ['dementia', 'mri', 'healthcare', 'UCSF'],
       tensors: 6
     },
     {
-      title: 'Diabetes Study 01.289.301',
+      name: 'Diabetes Study 01.289.301',
       description:
         'This was a double-blind diabetes study done in coordination with UC Santa Barbara between July 1st, 2017 and January 1st, 2019.',
       tags: ['diabetes', 'california', 'healthcare', 'UCSF', 'beekeeper'],
       tensors: 2
     },
     {
-      title: 'Dementia MRI Scans (10k)',
+      name: 'Dementia MRI Scans (10k)',
       description:
         'Performed a database dump of exactly 10,000 patient records from our EMR. All patients were diagnosed with dementia within 10 years of the MRIs.',
       tags: ['dementia', 'mri', 'healthcare', 'UCSF'],
       tensors: 6
     }
   ]
 
-  return (
-    <main className="space-y-4">
-      <div className="flex flex-col-reverse items-start space-y-4 space-y-reverse md:space-y-0 md:flex-row md:justify-between">
-        <h1 className="pr-4 text-4xl leading-12">Datasets</h1>
-        <button className="btn" onClick={() => alert('Create new dataset')}>
-          New dataset
-        </button>
-      </div>
-      <p className="mt-3 mb-6 text-xl font-light text-gray-400">Manage all private data hosted in your node</p>
+  const DatasetsList = ({datasets}) => {
+    return (
+      <>
+        {datasets.length > 0 ? (
```

Comment:
Need to handle errors here. If `datasets` is `undefined` for some reason, this will crash the app.